maskrcnn_benchmark/modeling/poolers.py

```diff
@@ -77,10 +88,11 @@ def __init__(self, output_size, scales, sampling_ratio):
 
     def convert_to_roi_format(self, boxes):
         concat_boxes = cat([b.bbox for b in boxes], dim=0)
-        device, dtype = concat_boxes.device, concat_boxes.dtype
+        # device, dtype = concat_boxes.device, concat_boxes.dtype
         ids = cat(
             [
-                torch.full((len(b), 1), i, dtype=dtype, device=device)
+                torch.full_like(b.bbox[:, :1], i)
```

Comment:
If you don't object to the "see-through" into b.bbox to get the shape, I would use this pattern. It does create an extra view, but should not be terribly slow unless it's in a very fast loop.

Reply:
This looks slightly uglier, but this is fine I think!

Reply:
We could also have a script method that uses torch.full and specifies a shape.

Reply:
We could make this a bbox method (create_field?). The tracing isn't particular about that.

Reply:
Let's keep it like that for now, it's fine :-)